# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: media/webrtc/trunk/webrtc/modules/desktop_capture/BUILD.gn
# Commit: 3f0981f37c1c
# Full Hash: 3f0981f37c1c499ad52016e526239b390703ba2a
# Author: Dan Minor <dminor@mozilla.com>
# Date: 2019-04-17 21:47:29
# Regressor Bug: 1496359
# File Overlap Count: 1
# Description:
#   Bug 1496359 - Add PipeWire support to desktop capture; r=ng
#   
#   This is an import of upstream commit 318da51f99f91e3de1192f29d7f1824958f9f13e.
#   
#   Depends on D27367
# ==============================================================================

diff -r ceed4e0b3373 -r 3f0981f37c1c media/webrtc/trunk/webrtc/modules/desktop_capture/BUILD.gn
--- a/media/webrtc/trunk/webrtc/modules/desktop_capture/BUILD.gn	Mon Apr 15 17:12:31 2019 +0000
+++ b/media/webrtc/trunk/webrtc/modules/desktop_capture/BUILD.gn	Mon Apr 15 17:26:27 2019 +0000
@@ -6,6 +6,7 @@
 # in the file PATENTS.  All contributing project authors may
 # be found in the AUTHORS file in the root of the source tree.
 
+import("//build/config/linux/pkg_config.gni")
 import("//build/config/ui.gni")
 import("../../webrtc.gni")
 
@@ -154,6 +155,23 @@
   }
 }
 
+if (is_linux) {
+  if (rtc_use_pipewire) {
+    pkg_config("pipewire") {
+      packages = [ "libpipewire-0.2" ]
+
+      defines = [ "WEBRTC_USE_PIPEWIRE" ]
+    }
+
+    pkg_config("gio") {
+      packages = [
+        "gio-2.0",
+        "gio-unix-2.0",
+      ]
+    }
+  }
+}
+
 rtc_source_set("desktop_capture") {
   public_deps = [
     ":desktop_capture_generic",